3_javascript/javascript2/week1/Find and count the Danish letters/script.js

```diff
@@ -0,0 +1,31 @@
+function danishCounter(string) {
+  let aECounter = 0;
+  let aOCounter = 0;
+  let oCounter = 0;
+
+  for (let index = 0; index < string.length; index++) {
+    const element = string[index];
+    if (element === "æ") {
+      aECounter++;
+    } else if (element === "å") {
+      aOCounter++;
+    } else if (element === "ø") {
+      oCounter++;
+    }
```

Comment:
Hi Magda!  I was trying but I couldn't find the way on how to obtain a more optimal solution. Before I completely melt my brain 🫠 , could I ask you for some clue as to which path I could take? 🙂 thanks !!

Reply:
sure! 
So what you did is that you assigned vowels to separate variables. But what if it was one variable, that is an object instead? Like so:
```
const vowels = { ae: 0, oe: 0, ao: 0, total: 0}
```
assuming this, you can iterate on the string like you do (or using array methods ;) ), but instead of checking for every vowel, you can see if a given key of your `vowels` object exists in the string and how many! You can access the key for example using `Object.keys` method (please look it up) or there are many others. So the flow would be as follows:
1. is the character === any of the keys in my `vowels`
2. if yes, +=1 value next to this key
3. += 1 to the total

Meraj had a very nice solution to this, but I would encourage you to try yourself before him or I give you the solution :)

Reply:
Super, I found an alternative, however it is not something I can build or recreate in my mind for now, I had to search a lot and it was not as spontaneous as perhaps with other tasks.

Thanks for your help :)